app/src/main/java/com/example/techstacksetup/MainActivity.java

```diff
@@ -34,31 +35,46 @@ protected void onCreate(Bundle savedInstanceState) {
         nextbtn = (Button)findViewById(R.id.nextBtn);
         helloNameInput  = (EditText)findViewById(R.id.editTextTextPersonName);
         helloName = (TextView)findViewById(R.id.helloName);
+        label= (TextView) findViewById(R.id.warning);
         final long[] mLastClickTime = {0};
 
+        helloNameInput.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View v) {
+                helloNameInput.setText("");
+            }
+        });
+
 
         nameButton.setOnClickListener(
                 new View.OnClickListener()
                 {
-                    public void onClick(View view)
-                    {
-                        if (SystemClock.elapsedRealtime() - mLastClickTime[0] < 10000){
-                            helloName.setText("Button cool down. Please do not spam the save button.");
-                            return;
+                    public void onClick(View view) {
+                        if (!helloNameInput.getText().toString().isEmpty()) {
```

Comment:
This if statement isn't necessary if the if statement on line 73 is placed before here. Just add a return to that if statement.

Reply:
this if statement is for spamming the save button, the if statement on line 73 is if you try to save a blank name it won't allow you to continue